Source/WebKit/GPUProcess/graphics/WebGPU/RemoteAdapter.cpp

```diff
@@ -52,6 +52,11 @@ RemoteAdapter::RemoteAdapter(PAL::WebGPU::Adapter& adapter, WebGPU::ObjectHeap&
 
 RemoteAdapter::~RemoteAdapter() = default;
 
+void RemoteAdapter::destruct()
+{
+    m_objectHeap.removeObject(m_identifier);
```

Comment:
Might want to add a comment after this line saying something like "beware: `this` is deleted now"